lightningd/invoice.c

```diff
@@ -806,8 +802,9 @@ static struct command_result *json_decodepay(struct command *cmd,
 	json_add_u64(response, "created_at", b11->timestamp);
 	json_add_u64(response, "expiry", b11->expiry);
 	json_add_pubkey(response, "payee", &b11->receiver_id);
-        if (b11->msatoshi)
-                json_add_u64(response, "msatoshi", *b11->msatoshi);
+        if (b11->msat)
+                json_add_amount_msat(response, *b11->msat,
+				     "msatoshi", "amount_msat");
```

Comment:
petition to rename "amount_msat" as "formatted_msat" or "formatted_value" or something to denote the 'formatted' i.e. string nature of the field value -- amount imo connotes an int or float type field.

Reply:
The problem is that we may eventually deprecate the old raw-numeric fields.  So it does need some semantic :(